fsspec/implementations/dropboxdrivefs.py

```diff
@@ -0,0 +1,239 @@
+import requests
+import dropbox
+from ..spec import AbstractFileSystem, AbstractBufferedFile
+from ..caching import AllBytes
+
+
+class DropboxDriveFileSystem(AbstractFileSystem):
+    """ Interface dropbox to connect, list and manage files
+    Parameters:
+    ----------
+    token : str
+          Generated key by adding a dropbox app in the user dropbox account.
+          Needs to be done by the user
+
+    """
+
+    def __init__(self, token, *args, **storage_options):
+        super().__init__(token=token, *args, **storage_options)
+        self.token = token
+        self.kwargs = storage_options
+        self.connect()
+
+    def connect(self):
+        """ connect to the dropbox account with the given token
+        """
+        self.dbx = dropbox.Dropbox(self.token)
+        self.session = requests.Session()
+        self.session.auth = ("Authorization", self.token)
+
+    def ls(self, path, detail=True, **kwargs):
+        """ List objects at path
+        """
+        while "//" in path:
+            path = path.replace("//", "/")
+        list_file = []
+        list_item = self.dbx.files_list_folder(
+            path, recursive=True, include_media_info=True
+        )
+        items = list_item.entries
+        while list_item.has_more:
+            list_item = self.dbx.files_list_folder_continue(list_item.cursor)
+            items = list_item.entries + items
+
+        if detail:
```

Comment:
Noting that this does not cache listings for future calls

Reply:
I try to look how it is done in other implementations but I don't see any difference... Can you give an hint about what you were expecting here please ? 
Thanks :)

Reply:
Some implementations populate a `dirache` attribute, which is a dict of {path: [file, file]} storing the listings that have been seen so far. It has proven tricky to use, but potentially speeds up repeat listings of the same directory. You are fine to proceed without it, but may get users later complaining that calls with `glob` and `walk` are slow.